### PR TITLE
Always allow current user to get own profile

### PR DESCRIPTION
**Pull request by chrisvire**
* getting users can get filtered by the current organization (which
  the user might not be in).

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
Needs a test :)

**Comment by chrisvire** [comment]
@NullVoxPopuli: Actually, I had just changed the title to include "WIP" since I decided I should add a test.  Well, I should have written the failing test first... :-)

File: source/SIL.AppBuilder.Portal.Backend.Tests/Acceptance/APIControllers/UsersTest.cs [hunk 1]
```diff
@@ -118,5 +120,26 @@ public async Task Patch_SomeUser_Different_Organization_NotFound()
...
+        public async Task Get_CurrentUser_Different_Organization_Found()
+        {
+            var user = await GetCurrentUser();
+
+            NeedsTestData<AppDbContext, Organization>(new List<Organization>
+            {
+                new Organization { Name = "Foo"}
+            });
+
+            var response = await Get($"/api/users/{user.Id}", "Foo");
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Backend.Tests/Acceptance/APIControllers/UsersTest.cs]: I believe the org second arg should be an org id. Even though

Comment (chrisvire) [reply source/SIL.AppBuilder.Portal.Backend.Tests/Acceptance/APIControllers/UsersTest.cs]: Ah yes...

File: source/SIL.AppBuilder.Portal.Backend.Tests/Acceptance/APIControllers/UsersTest.cs [hunk 2]
```diff
@@ -118,5 +120,26 @@ public async Task Patch_SomeUser_Different_Organization_NotFound()
...
+
+        /// <summary>
+        /// Getting the current user (edit the user profile) when a different
+        /// organization is the current organization (e.g. viewing users from
+        /// a different organization) should always be allowed and return the
+        /// current user.
+        /// </summary>
+        /// <returns>The current user different organization found.</returns>
+        [Fact]
+        public async Task Get_CurrentUser_Different_Organization_Found()
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Backend.Tests/Acceptance/APIControllers/UsersTest.cs]: This test name should be more meaningful.
maybe something like `Get_User_WhenIdMatchesTheCurrentUser_OrganizationIdDoesNotMatter()` or something to that effect.

Comment (chrisvire) [reply source/SIL.AppBuilder.Portal.Backend.Tests/Acceptance/APIControllers/UsersTest.cs]: `Get_CurrentUser_CurrentOrganizationDoesNotMatter_Found`

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Backend.Tests/Acceptance/APIControllers/UsersTest.cs]: it's not the CurrentUser endpoint though.
I guess I've (maybe) been doing

`{HTTP Verb}_{Action}_{Context?}_{Assertion}`

**Pull request merged**